File: testers/avro/java-quarkus/generated-and-checked-in/com/example/events/DynamicValueListener.java
package com.example.events;

import io.smallrye.mutiny.Uni;
import org.eclipse.microprofile.reactive.messaging.Incoming;
import org.eclipse.microprofile.reactive.messaging.Message;
import org.eclipse.microprofile.reactive.messaging.Metadata;

/** Event listener interface for dynamic-value topic. Implement this interface to handle events. */
public interface DynamicValueListener {
  /** Receive and dispatch events to handler methods */
  @Incoming("dynamic-value")
  default Uni<Void> receive(Message<Object> record) {
    return switch (record.getPayload()) {
      case null -> onUnknown(record);
      case DynamicValue e -> onDynamicValue(e, record.getMetadata());
      default -> onUnknown(record);
    };
  }

  /** Handle DynamicValue event */
  Uni<Void> onDynamicValue(DynamicValue event, Metadata metadata);

  /** Handle unknown event types. Override to customize behavior. */
  default Uni<Void> onUnknown(Message<Object> record) {
    return Uni.createFrom().voidItem();
  }
}

File: testers/avro/kotlin-quarkus-mutiny/generated-and-checked-in/com/example/events/DynamicValueListener.kt
package com.example.events

import io.smallrye.mutiny.Uni
import java.lang.Void
import org.eclipse.microprofile.reactive.messaging.Incoming
import org.eclipse.microprofile.reactive.messaging.Message
import org.eclipse.microprofile.reactive.messaging.Metadata

/** Event listener interface for dynamic-value topic. Implement this interface to handle events. */
interface DynamicValueListener {
  /** Handle DynamicValue event */
  abstract fun onDynamicValue(
    event: DynamicValue,
    metadata: Metadata
  ): Uni<Void>

  /** Handle unknown event types. Override to customize behavior. */
  fun onUnknown(record: Message<Any>): Uni<Void> {
    return Uni.createFrom().voidItem()
  }

  /** Receive and dispatch events to handler methods */
  @Incoming("dynamic-value")
  fun receive(record: Message<Any>): Uni<Void> {
    return when (val __r = record.getPayload()) {
      null -> onUnknown(record)
      is DynamicValue -> { val e = __r as DynamicValue; onDynamicValue(e, record.getMetadata()) }
      else -> onUnknown(record)
    }
  }
}